packages/devextreme/js/__internal/ui/calendar/m_calendar.ts

```diff
@@ -639,70 +681,72 @@ class Calendar<
     }
   }
 
-  _getMinDate() {
-    // eslint-disable-next-line @typescript-eslint/naming-convention
-    const _rangeMin = this.option('_rangeMin');
-    if (_rangeMin) {
-      return _rangeMin;
+  _getMinDate(): Date {
+    const { _rangeMin: rangeMin } = this.option();
+    if (rangeMin) {
+      return rangeMin;
     }
 
     if (this.min) {
       return this.min;
     }
-    // @ts-expect-error ts-error
-    this.min = this._dateOption('min') || new Date(1000, 0);
-    return this.min;
+
+    if (isDate(this.min)) {
+      return this.min;
+    }
```

Comment:
Are u sure the behavior is the same now? We stopped updating this.min + new condition "isDate"
And the same question to the next method below

Reply:
ur right, changes looks bad, thanks for pointing on this. refactored https://github.com/DevExpress/DevExtreme/pull/30599/commits/f8f2bf24138a6ed27b92dceaf4ad4b556f536c90